app/core/mutations/deleteUploadedImage.ts

```diff
@@ -0,0 +1,25 @@
+import { Storage } from '@google-cloud/storage'
+import { resolver } from 'blitz'
+import { z } from 'zod'
+
+import { gcpStorageBucketName } from 'app/core/environment'
+
+import { gcpStorageUrl } from '../constants'
+import { gc } from '../utils/gcpStorage'
+
+const DeleteUploadedImage = z.object({
+  url: z.string().nonempty(),
+})
+
+const deleteUploadedImage = resolver.pipe(
+  resolver.zod(DeleteUploadedImage),
+  resolver.authorize(),
+  async ({ url }) => {
+    const bucket = gc.bucket(gcpStorageBucketName)
+    const file = bucket.file(url.slice(gcpStorageUrl.length + 1))
+
+    await file.delete()
```

Comment:
Should be a way to enforce that the user can only delete their own files, if it's not too late

Reply:
See #43

Reply:
Not done yet in #43 though